todo.py

```diff
@@ -5,9 +5,24 @@ def __init__(self):
         self.tasks = []
 
     def add_task(self, task):
-        """Adds a new task to the list."""
-        self.tasks.append(task)
-        print(f'Task added: {task}')
+        """Adds a new task to the list if it doesn't already exist."""
+        if self.task_exists(task):
+            print(f"Task '{task}' already exists in the list! Cannot add duplicate tasks.")
+        else:
+            self.tasks.append(task)
+            print(f'Task added: {task}')
+
+    def task_exists(self, task):
+        """Checks if a task already exists (case insensitive)."""
```

Comment:
It could also check for trailing whitespace and ignore that so 'task' and 'task ' would also be flagged

Reply:
Good idea! Done